pkg/resource/providerconfig.go

```diff
@@ -46,6 +56,119 @@ func IsMissingReference(err error) bool {
 	return ok
 }
 
+// A CredentialExtractor extracts a provider's credentials.
+type CredentialExtractor interface {
+	Extract(context.Context, string, interface{})
+}
+
+// An ExtractionHandlerFn is a function that can extract a specific type of
+// credentials.
+type ExtractionHandlerFn func(ctx context.Context, c interface{}) ([]byte, error)
+
+// EnvLookupFn looks up an environment variable.
+type EnvLookupFn func(string) string
+
+// OsEnvLookup looks up an environment variable from the OS.
+func OsEnvLookup(s string) string {
+	e, _ := os.LookupEnv(s)
+	return e
+}
```

Comment:
As far as I can tell this function is identical to `os.Getenv`. Could we use that instead?

Reply:
It serves the exact same purpose, just makes it more testable. I could be convinced to remove it and set the env var in the test directly if desired.

Reply:
Either way, it should be updated to just return `os.GetEnv` rather than using `os.LookupEnv`, but I like the quality of wrapping it nonetheless.

Reply:
Good point, didn't see that until I updated to `os.Getenv` from `os.LookupEnv` :+1: